.github/workflows/ci-dgraph-code-coverage.yml

```diff
@@ -0,0 +1,75 @@
+name: ci-dgraph-code-coverage
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+      - ready_for_review
+    branches:
+      - main
+  schedule:
+    - cron: "0 22 * * *"
+jobs:
+  dgraph-code-coverage:
+    # TODO: uncomment this after release work
+    # if: github.event.pull_request.draft == false
+    runs-on: [self-hosted, x64]
```

Comment:
@anurags92 - thinking about this again ... I think we should have this embedded in `ci-dgraph-tests`. The additional time we take to run coverage collection and push is may be 1-2mins. The issue with keeping a new pipeline will cause more resource consumptions because of dual test-runs.

Could we shrink this to what we have?

Reply:
The extra time is one thing, but the binary you generate for collecting coverage is not the same binary which the production build would runs. That's another reason I felt we can keep the things separate and clean. I don't have a scenario in which production binary and coverage binary differ in results, but I feel safer running the same binary in tests as the production dgraph clusters.

Reply:
yeah you are right !!! I remember discussing this as well .. apologies! resolving this comment.